MIT-LCP/physionet-build
# OAuth Provider Implementation

This PR adds an [OAuth 2.0](https://oauth.net/2/) Provider. The Provider can be tested following these [procedures](https://www.toolsqa.com/postman/oauth-2-0-authorization-with-postman/).

Testing and Usage Instructions :

 1. Create an Application from the admin panel to get the client-id & client-secrets
![image](https://github.com/MIT-LCP/physionet-build/assets/46196557/0275e9a4-6697-4f91-9e5e-9e28e9790750)
(Also put in the redirect-uri for Postman : https://www.getpostman.com/oauth2/callback)

2. Fill in the details in the postman
![image](https://github.com/MIT-LCP/physionet-build/assets/46196557/73f661ce-5c51-4fbf-986f-5b038804961a)

For a detailed version - check out the procedures mentioned [here](https://www.toolsqa.com/postman/oauth-2-0-authorization-with-postman/).

**Usage Scenarios** :

This enables the Platform to be a provider and provide API-based resource authorization using the [OAuth](https://oauth.net/2/) workflow. This can enable Physionet-based authentication for third-party users. 
For Eg: Someone who is not registered with HDN but is registered with Physionet - can use this functionality(once HDN is added as a client) to access a pre-determined resource. This functionality needs to be extended for the implementation of these scenarios and is a first step towards it.
